internal/session/service_authorize_connection.go

```diff
@@ -0,0 +1,27 @@
+package session
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/errors"
+)
+
+// AuthorizeConnection is a domain service function that will:
+// * create a connection, if the session has not reached its connection limit and the session is not expired and is still active
+// * Return the connection, its states, and an AuthzSummary.
+// If authorization fails, it returns an error with Code InvalidSessionState.
```

Comment:
maybe something like this to highlight the conditions it uses?
```suggestion
// AuthorizeConnection is a domain service function that will create a Connection
// for a session if the following criteria are met:
// * The session is active.
// * The session is not expired.
// * The session has not reached its connection limit or has a connection limit of -1.
// If any of these criteria is not met, it returns an error with Code InvalidSessionState.